### PR TITLE
Configure default for Git options and display in summary

### LINKED ISSUE #166: Expand `-o` / `--git-options` functionality
## Description

A new option `-o` / `--git-options` should be made available to allow users to provide Git options (e.g. depth). 

### Non-interactive mode

- `--git-options` is not a mandatory argument when invoking the script from the command line
- If `--git-options` is not specified from the command line, use `--depth=1` as the value passed to Git when cloning
- In the experiment summary, always print the final Git options (similar to the branch name, commit id, etc.)

### Interactive mode

- If `--git-options` is passed from the command line, use the specified value as the default shown to the user when querying the user for the Git options
- If `--git-options` is not passed from the command line, use `--depth=1` as the default shown to the user when querying the user for the Git options
- In the experiment summary, always print the final Git options (similar to the branch name, commit id, etc.)
- In the command line invocation summary, only include the `--git-options` if the final Git options are different from the default, i.e. different from `--depth=1`

### PR DESCRIPTION
**Pull request by erichaagdev**
## Non-interactive mode

✅ `--git-options` is not a mandatory argument when invoking the script from the command line

**When** 

```
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble
```

**Then experiment finishes successfully**

image

---

✅ If `--git-options` is specified from the command line, pass the value to Git when cloning

**When**

```
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --git-options --depth=1\ --shallow-submodules
  --debug
```

**Then Git options are used**

image

**And**

image

---

✅ If `--git-options` is not specified from the command line, use `--depth=1` as the value passed to Git when cloning

**When** 

```
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --debug
```

**Then `--depth=1` can be seen as the only option in the clone**

image

---

✅ In the experiment summary, always print the final Git options (similar to the branch name, commit id, etc.)

**When**

```
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble
```

**Then summary includes Git options**

image

## Interactive mode

✅ If `--git-options` is passed from the command line, use the specified value as the default shown to the user when querying the user for the Git options

**When**

```shell
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --git-options --depth=1\ --shallow-submodules \
  --interactive
```

**Then Git options from command line is shown as default**

image

---

✅ If `--git-options` is not passed from the command line, use `--depth=1` as the default shown to the user when querying the user for the Git options

**When**

```
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --interactive
```

**Then default is used**

image

---

✅ In the experiment summary, always print the final Git options (similar to the branch name, commit id, etc.)

**When**

```shell
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --interactive
```

**Then default is shown in summary**

image

**When**

```shell
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --git-options --depth=1\ --shallow-submodules \
  --interactive
```

**Then Git options is shown in summary**

image

---

✅ In the command line invocation summary, only include the `--git-options` if the final Git options are different from the default, i.e. different from `--depth=1`

**When**

```shell
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --interactive
```

**Then no Git options are shown in CLI invocation summary**

image

**When**

```shell
./01-validate-incremental-building.sh \
  --git-repo git@github.com:gradle/gradle-enterprise-build-validation-scripts.git \
  --tasks assemble \
  --git-options --depth=1\ --shallow-submodules \
  --interactive
```

**Then Git options are shown in CLI invocation summary**

image

---

Closes #166

### CODE REVIEW

**Review by jthurne — APPROVED** [review]
LGTM

File: release/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-2.0.1
+2.1.0
```
Comment (erichaagdev) [inline release/version.txt:1]: I assume this is enough of a change to warrant a new minor release. If you don't agree or if this is not the right time to determine this please change it back.

**Review by jthurne — APPROVED** [review]
👍 Nice!

File: release/version.txt (hunk 1 above)
Comment (etiennestuder) [reply release/version.txt:1]: I will revert this on this branch to contina the scope of the PR. But we will adjust the version number for sure once we are ready to release.

File: components/scripts/lib/cli-parsers/common.m4 [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # ARG_OPTIONAL_BOOLEAN([enable-gradle-enterprise],[e],[])
 # ARG_OPTIONAL_SINGLE([git-branch],[b],[])
 # ARG_OPTIONAL_SINGLE([git-commit-id],[c],[])
-# ARG_OPTIONAL_SINGLE([git-options],[o],[])
+# ARG_OPTIONAL_SINGLE([git-options],[o],[],[--depth=1])
```
Comment (etiennestuder) [inline components/scripts/lib/cli-parsers/common.m4:6]: @erichaagdev @jthurne just curious: what impact does the declaration of `depth` on this line have?

**Review by etiennestuder — APPROVED** [review]
Looking good. Thanks for the summary of your verifications on the comment of the GH issue.

File: components/scripts/lib/cli-parsers/common.m4 (hunk 2 above)
Comment (jthurne) [reply components/scripts/lib/cli-parsers/common.m4:6]: That sets the default value for git-options (when the scripts are used non-interactively).

**Pull request merged**